main.c

```diff
@@ -45,24 +45,76 @@ char	*lexprog;	/* points to program argument if it exists */
 extern	int errorflag;	/* non-zero if any syntax errors; set by yyerror */
 enum compile_states	compile_time = ERROR_PRINTING;
 
-#define	MAX_PFILE	20	/* max number of -f's */
-
-char	*pfile[MAX_PFILE];	/* program filenames from -f's */
-int	npfile = 0;	/* number of filenames */
-int	curpfile = 0;	/* current filename */
+static char	**pfile;	/* program filenames from -f's */
+static size_t	maxpfile;	/* max program filename */
+static size_t	npfile;		/* number of filenames */
+static size_t	curpfile;	/* current filename */
 
 bool	safe = false;	/* true => "safe" mode */
 
+static __attribute__((__noreturn__)) void fpecatch(int n
+#ifdef SA_SIGINFO
+	, siginfo_t *si, void *uc
+#endif
+)
+ {
+#ifdef SA_SIGINFO
+	static const char *emsg[] = {
+	    [0] = "Unknown error",
+	    [FPE_INTDIV] = "Integer divide by zero",
+	    [FPE_INTOVF] = "Integer overflow",
+	    [FPE_FLTDIV] = "Floating point divide by zero",
+	    [FPE_FLTOVF] = "Floating point overflow",
+	    [FPE_FLTUND] = "Floating point underflow",
+	    [FPE_FLTRES] = "Floating point inexact result",
+	    [FPE_FLTINV] = "Invalid Floating point operation",
+	    [FPE_FLTSUB] = "Subscript out of range",
+	};
+#endif
+	FATAL("floating point exception"
+#ifdef SA_SIGINFO
+	    ": %s\n", (size_t)si->si_code < sizeof(emsg) / sizeof(emsg[0]) &&
+	    emsg[si->si_code] ? emsg[si->si_code] : emsg[0]
+#endif
+	    );
+}
+
 /* Can this work with recursive calls?  I don't think so.
 void segvcatch(int n)
 {
 	FATAL("segfault.  Do you have an unbounded recursive call?", n);
 }
 */
 
+static const char *
+setfs(char *p)
+{
+	/* wart: t=>\t */
+	if (p[0] == 't' && p[1] == '\0')
+		return "\t";
+	else
+	    if (p[0] != '\0')
```

Comment:
Please join lines 95 and 96 and reformat the whole function to K&R tab indented.

Reply:
Ok.